### PR TITLE
Feat wind indicators

### PR DESCRIPTION
**Pull request by tdivoll**
Getting close to having the wind indicators completed. Currently, indicators change based on each timepoint, but do not update with changes in the DateSlider. I know what needs to happen, just haven't figured out how to implement it. Need to add `windSpeed` and `windDirection` to the `watch()`
image

### CODE REVIEW

File: src/components/buoy-telemetry/BuoyMap.vue [hunk 1]
```diff
@@ -111,10 +265,36 @@ onMounted(() => {
...
+    for (var b = 0; b < windSpeedBins.length - 1; b++) {
+      map.value.addLayer({
+        id: "svgfile" + b,
+        type: "symbol",
+        filter: [
+          "all",
+          [">=", "windSpeed", windSpeedBins[b]],
+          ["<", "windSpeed", windSpeedBins[b + 1]],
+        ],
+        source: "points", // points.filter(annotatedwind)?
```
Comment (eldu) [inline src/components/buoy-telemetry/BuoyMap.vue:282]: taking out commented code
```suggestion
        source: "points",
```

File: src/components/buoy-telemetry/BuoyMap.vue [hunk 2]
```diff
@@ -145,6 +325,7 @@ const updateMarkers = () => {
         .addTo(map.value)
     );
   });
+  // annotatedWindSpeed.value.forEach(({ station_name }))
```
Comment (eldu) [inline src/components/buoy-telemetry/BuoyMap.vue:328]: Delete commented code

File: src/views/datasets/buoy-telemetry/summary.vue [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <DashboardCard width="full">
+  <DashboardCard width="two-thirds">
```
Comment (eldu) [inline src/views/datasets/buoy-telemetry/summary.vue:2]: Changes in this file don't seem to be related to this PR?

Comment (eldu) [reply src/views/datasets/buoy-telemetry/summary.vue:2]: What are these changes for?

File: src/components/buoy-telemetry/BuoyMap.vue [hunk 4]
```diff
@@ -111,10 +265,36 @@ onMounted(() => {
...
+          "icon-image": `wind-speed-${b + 1}`, //mapped value of wind
+          "icon-offset": [-100, -40], // optically centered
+          "icon-size": 0.3,
+          "icon-rotate": {
+            property: "windDirection",
+            stops: [
+              [0, 90],
+              [270, 360],
+            ],
+          },
```
Comment (eldu) [inline src/components/buoy-telemetry/BuoyMap.vue:294]: Question about the rotate! So, if the two buoys have the same windspeed and different winddirections will the different wind directions show? 

At the moment,  I'm realizing that the for loop at 273 goes through each wind direction bin. But I'm wondering if instead it should be looping through each buoy?

**Review by eldu — COMMENTED** [review]
Looking good Tim! I added in a question and some small comments !

**Review by eldu — CHANGES_REQUESTED** [review]
Meant to chose request changes

File: src/components/buoy-telemetry/BuoyMap.vue (hunk 1 above)
Comment (tdivoll) [reply src/components/buoy-telemetry/BuoyMap.vue]: deleted!

File: src/components/buoy-telemetry/BuoyMap.vue (hunk 2 above)
Comment (tdivoll) [reply src/components/buoy-telemetry/BuoyMap.vue:338]: deleted!

File: src/views/datasets/buoy-telemetry/summary.vue (hunk 3 above)
Comment (tdivoll) [reply src/views/datasets/buoy-telemetry/summary.vue:2]: ideally, it would have been a different PR, it is just to make the buoy image on it's own card

File: src/components/buoy-telemetry/BuoyMap.vue (hunk 4 above)
Comment (tdivoll) [reply src/components/buoy-telemetry/BuoyMap.vue]: it does rotate for wind indicators at each buoy, I believe through the "points"

File: src/components/buoy-telemetry/BuoyMap.vue [hunk 5]
```diff
@@ -35,42 +114,121 @@ const store = inject("store") as BuoyStore;
...
-    .filter((row) => row.variable === props.variable)
-    .map((row) => {
-      return {
-        ...row,
-        color: colorScale(row.value),
-        size: sqrtScale(row.value),
-      };
-    });
+const knotsPerMS = 1.94384;
+const windSpeedBins = [
```
Comment (eldu) [inline src/components/buoy-telemetry/BuoyMap.vue:130]: Maybe rename this variable so we know that this is in knots instead of just the comment?

File: src/components/buoy-telemetry/BuoyMap.vue [hunk 6]
```diff
@@ -81,53 +239,84 @@ onMounted(() => {
...
+      map.value.removeLayer("svgfile" + b);
+    }
+  }
+  if (map.value.getLayer("points")) {
+    map.value.removeLayer("points");
+  }
+  if (map.value.getSource("points")) {
+    map.value.removeSource("points");
+    map.value.addSource("points", geoJSON.value);
+  }
```
Comment (eldu) [inline src/components/buoy-telemetry/BuoyMap.vue:276]: I'd move the `addSource` outside of the if statement to parallel the structure of the rest of the block of code

```suggestion
  if (map.value.getSource("points")) {
    map.value.removeSource("points");
  }
map.value.addSource("points", geoJSON.value);
```

File: src/views/datasets/buoy-telemetry/summary.vue (hunk 3 above)
Comment (eldu) [reply src/views/datasets/buoy-telemetry/summary.vue:2]: Ahh okay!

**Review by eldu — CHANGES_REQUESTED** [review]
Looks good Tim! Some small style changes! 

Overall good things!

File: src/components/buoy-telemetry/BuoyMap.vue [hunk 7]
```diff
@@ -35,42 +114,121 @@ const store = inject("store") as BuoyStore;
...
+
+const annotatedWindSpeed = computed(() => {
+  if (props.showWind) {
+    return props.samples
+      .filter((row) => row.variable === "WindSpeedAverage")
+      .map((row) => {
+        return {
+          ...row,
+        };
+      });
```
[the remark this answers is not in the document]
Comment (tdivoll) [reply src/components/buoy-telemetry/BuoyMap.vue:165]: you are right! I couldn't get it to populate using `.includes()`, so copied the format from `annotatedChlorophyll` and we don't need the `.map()` for the wind.

File: src/components/buoy-telemetry/BuoyMap.vue [hunk 8]
```diff
@@ -35,42 +114,121 @@ const store = inject("store") as BuoyStore;
...
+
+const annotatedWindDir = computed(() => {
+  if (props.showWind) {
+    return props.samples
+      .filter((row) => row.variable === "WindDirectionFrom")
+      .map((row) => {
+        return {
+          ...row,
+        };
+      });
```
[the remark this answers is not in the document]
Comment (tdivoll) [reply src/components/buoy-telemetry/BuoyMap.vue:179]: fixed with the comment above

File: src/components/buoy-telemetry/BuoyMap.vue [hunk 9]
```diff
@@ -35,42 +114,121 @@ const store = inject("store") as BuoyStore;
...
+        ({ latitude, longitude, station_name }) => {
+          return {
+            type: "Feature",
+            geometry: {
+              type: "Point",
+              coordinates: [longitude, latitude],
+            },
+            properties: {
+              windSpeed:
+                annotatedWindSpeed.value.find((w) => {
```
[the remark this answers is not in the document]
Comment (tdivoll) [reply src/components/buoy-telemetry/BuoyMap.vue:216]: Good call, I changed it to `annotatedWindSpeedms`. Capitalizing the m felt wrong because meters (m) is always lowercase

File: src/components/buoy-telemetry/BuoyMap.vue (hunk 5 above)
Comment (tdivoll) [reply src/components/buoy-telemetry/BuoyMap.vue:130]: changed this to `windSpeedBinsKnots`

File: src/components/buoy-telemetry/BuoyMap.vue (hunk 6 above)
Comment (tdivoll) [reply src/components/buoy-telemetry/BuoyMap.vue:276]: good catch, thank you!

**Review by eldu — APPROVED** [review]
Overall I think this is good! Perhaps we can make an issue about the blurriness to address another time if that wasn't an easy fix !
